source/common/tracing/http_tracer_impl.cc

```diff
@@ -70,21 +69,21 @@ Decision HttpTracerUtility::isTracing(const StreamInfo::StreamInfo& stream_info,
     return {Reason::HealthCheck, false};
   }
 
-  if (!request_headers.RequestId()) {
+  Http::RequestIDExtensionSharedPtr rid_extension = stream_info.getRequestIDExtension();
+  if (rid_extension == nullptr) {
```

Comment:
This can't happen, right? Or can it happen earlier in server load? If so maybe comment? It seems like maybe this should actually be a function call into the extension to check for something and this might be a bug?